rskj-core/src/main/java/org/ethereum/core/BlockFactory.java

```diff
@@ -189,24 +221,31 @@ private BlockHeader decodeHeader(RLPList rlpHeader, boolean sealed) {
         int r = 15;
 
         int uncleCount = 0;
-        if (rlpHeader.size() == RLP_HEADER_SIZE || rlpHeader.size() == RLP_HEADER_SIZE_WITH_MERGED_MINING) {
-            byte[] ucBytes = rlpHeader.get(r++).getRLPData();
-            uncleCount = parseBigInteger(ucBytes).intValueExact();
-        }
+        byte[] ucBytes = rlpHeader.get(r++).getRLPData();
+        uncleCount = parseBigInteger(ucBytes).intValueExact();
```

Comment:
I must check if it ever happened in Mainnet, but this doesn't seem to catch the case where rlpHeader.size() == 15.

Reply:
The original file (in master branch) specifies only 2 valid sizes : 16 and 19.
Look at ; https://github.com/rsksmart/rskj/blob/f747100c327e72e8ab29f0ff9d53bfd5c7e95a1d/rskj-core/src/main/java/org/ethereum/core/BlockFactory.java#L143-L148

As I've added the partitionEnds fields, now 17 and 20 are allowed too.

Also I've removed the test on line 192 because I though it was redondant with the one at the beginning of the method (line 143 on master, see above),
https://github.com/rsksmart/rskj/blob/f747100c327e72e8ab29f0ff9d53bfd5c7e95a1d/rskj-core/src/main/java/org/ethereum/core/BlockFactory.java#L192